packages/runtime/src/ax.tsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * Joins classes together and ensures atomic declarations of a single group exist.
+ * This function is made to be fast and small - so it takes some liberties with its API.
+ *
+ * Atomic declarations take the form of `_{group}{value}`,
+ * where both `group` and `value` are hashes **four characters long**.
+ *
+ * Input:
+ *
+ * ```
+ * ax(['_aaaabbbb', '_aaaacccc'])
+ * ```
+ *
+ * Output:
+ *
+ * ```
+ * '_aaaacccc'
+ * ```
+ *
+ * @param classes
+ */
+export const ax = (classes: (string | undefined | false)[]): string => {
+  const found: Record<string, string> = {};
+
+  for (let i = 0; i < classes.length; i++) {
+    const cls = classes[i];
+    if (!cls) {
+      continue;
+    }
+
+    const group = cls.slice(0, cls.charCodeAt(0) === 95 ? 5 : undefined);
```

Comment:
`95` is underscore right? Got confused at first 😅

Reply:
Are we assuming `cls ` should always have length of `8`?

Reply:
no it can be more - we just make assumptions that we'll take a slice of it if it starts with underscore

Reply:
>  * Atomic declarations take the form of `_{group}{value}`, * where both `group` and `value` are hashes **four characters long**.

This gave me that impression. No worries 😄

Reply:
ah maybe we can rephrase